recipes/SLURP/direct/hparams/train_with_wav2vec2.yaml

```diff
@@ -0,0 +1,183 @@
+# ############################################################################
+# Model: Direct SLU
+# Encoder: Pre-trained wav2vec2 -> LSTM
+# Decoder: GRU + beamsearch
+# Tokens: BPE with unigram
+# losses: NLL
+# Training: SLURP
+# Authors:  Loren Lugosch, Mirco Ravanelli 2020 Boumadane Abdelmoumene, AbdelWahab Heba, Yingzhi Wang 2021
+# ############################################################################
+
+# Seed needs to be set at top of yaml, before objects with parameters are made
+seed: 1986
+__set_seed: !apply:torch.manual_seed [!ref <seed>]
+output_folder: !ref results/<seed>
+save_folder: !ref <output_folder>/save
+train_log: !ref <output_folder>/train_log.txt
+
+# Data files
+# The SLURP dataset will be automatically downloaded in the specified data_folder
+data_folder: !PLACEHOLDER # e.g, /localscratch/SLURP
+data_folder_rirs: !ref <data_folder>
+train_splits: ["train_synthetic", "train_real"]
+csv_train: !ref <output_folder>/train-type=direct.csv
+csv_valid: !ref <output_folder>/devel-type=direct.csv
+csv_test: !ref <output_folder>/test-type=direct.csv
+tokenizer_file: https://www.dropbox.com/s/w72vtnf273yjwu4/SLURP_58_unigram.model?dl=1
+skip_prep: False
+
+
+# URL for the wav2vec2 model, you can change to benchmark diffrenet models
+wav2vec2_hub: "facebook/hubert-base-ls960"
+
+# Training parameters
+number_of_epochs: 35
+batch_size: 2
+lr: 0.0003
+lr_wav2vec2: 0.00001
+token_type: unigram # ["unigram", "bpe", "char"]
+sorting: random
+ckpt_interval_minutes: 15 # save checkpoint every N min
+
+#freeze all wav2vec2
+freeze_wav2vec2: False
+
+#set to true to freeze the CONV part of the wav2vec2 model
+freeze_wav2vec2_conv: True
+
+# Model parameters
+sample_rate: 16000
+emb_size: 128
+dec_neurons: 512
+output_neurons: 58 # index(eos/bos) = 0
+encoder_dim: 768
+
+# Decoding parameters
+bos_index: 0
+eos_index: 0
+min_decode_ratio: 0.0
+max_decode_ratio: 10.0
+slu_beam_size: 80
+eos_threshold: 1.5
+temperature: 1.25
+
+dataloader_opts:
+    batch_size: !ref <batch_size>
+    shuffle: True
+
+epoch_counter: !new:speechbrain.utils.epoch_loop.EpochCounter
+    limit: !ref <number_of_epochs>
+
+# Models
+
+wav2vec2: !new:speechbrain.lobes.models.huggingface_wav2vec.HuggingFaceWav2Vec2
+    source: !ref <wav2vec2_hub>
+    output_norm: True
+    freeze: !ref <freeze_wav2vec2>
+    save_path: !ref <save_folder>/wav2vec2_checkpoint
+
+output_emb: !new:speechbrain.nnet.embedding.Embedding
+    num_embeddings: !ref <output_neurons>
+    embedding_dim: !ref <emb_size>
+
+dec: !new:speechbrain.nnet.RNN.AttentionalRNNDecoder
+    enc_dim: !ref <encoder_dim>
+    input_size: !ref <emb_size>
+    rnn_type: lstm
+    attn_type: content
+    hidden_size: !ref <dec_neurons>
+    attn_dim: 512
+    num_layers: 3
+    scaling: 1.0
+    dropout: 0.0
+
+seq_lin: !new:speechbrain.nnet.linear.Linear
+    input_size: !ref <dec_neurons>
+    n_neurons: !ref <output_neurons>
+
+env_corrupt: !new:speechbrain.lobes.augment.EnvCorrupt
```

Comment:
`env_corrupt` is not used ? 
please consider removing it !

I'm wondering why you are not using it ? is it related to GPU memory ? (we use the RTX 8000 48Gb, is your model got OOM in this case ?)

Reply:
I didn't try it but maybe it's a good point, @Moumeneb1 you got OOM with it ?

Reply:
Yes, I was running out of RAM due to the batch size being doubled. I felt it would be more interesting to move larger batches with better gradients than smaller batches with some env corruption. If we get larger gpus we can try it.

Reply:
Hey @Moumeneb1 , @BenoitWang , I'm running an exp right now with:
1- 4 GPU V10032Gb
2- Batch of 2 with (env_corrupt + augmentation) total -> 4
3- Grad accumulation of 4
The exp took 1H50min per epoch, let me report the results tomorrow..

Reply:
Thanks @Moumeneb1, so for now I just remove it. @aheba Ready from our side.